modules/core/src/main/java/org/apache/ignite/internal/visor/ru/VisorRollingUpgradeStatus.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Creates a wrapper class that represents a value of {@link RollingUpgradeStatus}.
+     *
+     * @param enabled {@code true} if Rolling Upgrade is enabled.
+     * @param forcedModeEnabled {@code true} if forced mode is enabled.
+     * @param initVer Initial version.
+     * @param targetVer Resulting version.
+     * @param supportedFeatures Feature set that is supported by nodes.
+     */
+    public VisorRollingUpgradeStatus(
+        boolean enabled,
+        boolean forcedModeEnabled,
+        String initVer,
+        String targetVer,
+        Set<String> supportedFeatures
+    ) {
+        this.enabled = enabled;
+        this.forcedModeEnabled = forcedModeEnabled;
+        this.initVer = initVer;
+        this.targetVer = targetVer;
+        this.supportedFeatures = supportedFeatures;
+    }
+
+    /**
+     * Creates a wrapper class that represents a value of {@link RollingUpgradeStatus}.
+     *
+     * @param status Rolling upgrade status to be copied.
+     */
+    public VisorRollingUpgradeStatus(RollingUpgradeStatus status) {
+        this(
+            status.enabled(),
+            status.forcedModeEnabled(),
+            status.initialVersion().toString(),
+            (status.targetVersion() != null)? status.targetVersion().toString(): "",
+            status.supportedFeatures().stream().map(Enum::name).collect(Collectors.toSet()));
+    }
+
+    /**
+     * Returns {@code true} if Rolling Upgrade is enabled and is in progress.
+     *
+     * @return {@code true} if Rolling Upgrade is enabled.
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * @return {@code true} if strict mode is disabled.
+     */
+    public boolean isForcedModeEnabled() {
+        return forcedModeEnabled;
+    }
+
+    /**
+     * Returns the version that is used as starting point for Rolling Upgrade.
+     *
+     * @return Initial version.
+     */
+    public String getInitialVersion() {
+        return initVer;
+    }
+
+    /**
+     * Returns the target version.
+     * The returned value can be {@code null} if Rolling Upgrade is not in progress
+     * or target version is not determined yet.
+     *
+     * This method makes sense only for the case when the {@code forced} mode is disabled.
+     *
+     * @return Target version.
+     */
+    public String getTargetVersion() {
+        return targetVer;
+    }
+
+    /**
+     * Returns a set of features that is supported by all nodes in the cluster.
+     *
+     * @return Feature set supported by all cluster nodes.
+     */
+    public Set<String> getSupportedFeatures() {
+        return supportedFeatures;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void writeExternalData(ObjectOutput out) throws IOException {
+        out.writeBoolean(enabled);
+        out.writeBoolean(forcedModeEnabled);
+        U.writeString(out, initVer);
+        U.writeString(out, targetVer);
+        out.writeObject(supportedFeatures);
```

Comment:
U.writeCollection can be used for collection object.

Reply:
Thanks, fixed.